src/entities/swap/swaps/v3/index.ts

```diff
@@ -97,30 +99,44 @@ export class SwapV3 implements SwapBase {
 
         return this.isBatchSwap
             ? this.queryBatchSwap(client, block)
-            : this.querySingleSwap(client, block);
+            : this.querySingleSwap(client, block, account);
     }
 
     private async querySingleSwap(
         client: PublicClient,
         block?: bigint,
+        account?: Address,
     ): Promise<ExactInQueryOutput | ExactOutQueryOutput> {
         const routerContract = getContract({
             address: BALANCER_ROUTER[this.chainId],
             abi: balancerRouterAbi,
             client,
         });
         if ('exactAmountIn' in this.swaps) {
-            const { result } =
-                await routerContract.simulate.querySwapSingleTokenExactIn(
-                    [
-                        this.swaps.pool,
-                        this.swaps.tokenIn,
-                        this.swaps.tokenOut,
-                        this.swaps.exactAmountIn,
-                        DEFAULT_USERDATA,
-                    ],
-                    { blockNumber: block },
-                );
+            const { result } = account
+                ? await routerContract.simulate.swapSingleTokenExactIn(
+                      [
+                          this.swaps.pool,
+                          this.swaps.tokenIn,
+                          this.swaps.tokenOut,
+                          this.swaps.exactAmountIn,
+                          0n, // minAmountOut
```

Comment:
Same problem with `minAmountOut`

Reply:
This should be fine (see [Router](https://github.com/balancer/balancer-v3-monorepo/blob/e35fecb6634f7876f6c7b267bbf0b0e38bc008a6/pkg/vault/contracts/Router.sol#L1063) which does the same).